RecoLocalTracker/SiPixelClusterizer/plugins/alpaka/SiPixelRawToCluster.cc

```diff
@@ -255,7 +388,9 @@ namespace ALPAKA_ACCELERATOR_NAMESPACE {
                                   fedCounter,
                                   useQuality_,
                                   includeErrors_,
-                                  verbose_);
+                                  digiMorphingConfig_,
+                                  morphingModulesDevice_->data(),
```

Comment:
If `morphingModulesHost` is empty (line [286](https://github.com/cms-sw/cmssw/pull/48734/files#diff-c6b1691f372e5ef007c54b1eb6ebd270a6ee49c0309ddd3460a065b25508ded9R286)), `morphingModulesDevice_` is reset to a non-initialised `optional` object, and calling `->data()` on it is at best undefined behaviour.

Rather than `reset()` can you initialise it to an empty buffer ?

---

Can you run some test to make sure the code works correctly even when there are no modules enabled for the digi morphing ?